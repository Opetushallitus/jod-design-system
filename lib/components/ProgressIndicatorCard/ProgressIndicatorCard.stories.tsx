import type { StoryObj } from '@storybook/react-vite';
import type { TitledMeta } from '../../utils';

import { ProgressIndicatorCard } from './ProgressIndicatorCard';

const meta = {
  title: 'Cards/ProgressIndicatorCard',
  component: ProgressIndicatorCard,
  tags: ['autodocs'],
} satisfies TitledMeta<typeof ProgressIndicatorCard>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  decorators: [
    (Story) => (
      <div style={{ padding: '20px' }}>
        <Story />
      </div>
    ),
  ],
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/design/6M2LrpSCcB0thlFDaQAI2J/cx_jod_client?node-id=3857-2323',
    },
    docs: {
      description: {
        story: 'Showing the progress indicator card',
      },
    },
    layout: 'fullscreen',
    viewport: {
      defaultViewport: 'desktop',
    },
  },
  args: {},
};
