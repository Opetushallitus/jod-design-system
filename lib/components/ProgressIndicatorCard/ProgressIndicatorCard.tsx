import { Spinner } from '../Spinner/Spinner';

export const ProgressIndicatorCard = () => {
  return (
    <div className="ds:shadow-border ds:rounded ds:bg-bg-gray-2-25 ds:min-h-[285px] ds:flex ds:items-center ds:justify-center">
      <Spinner size={64} color="accent" className="ds:absolute" />
    </div>
  );
};
