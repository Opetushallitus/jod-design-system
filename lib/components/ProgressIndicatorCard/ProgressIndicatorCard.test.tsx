import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { expect, test } from 'vitest';
import { ProgressIndicatorCard } from './ProgressIndicatorCard';

test('renders ProgressIndicatorCard', () => {
  const { container } = render(<ProgressIndicatorCard />);
  expect(container.firstChild).toMatchSnapshot();
});
