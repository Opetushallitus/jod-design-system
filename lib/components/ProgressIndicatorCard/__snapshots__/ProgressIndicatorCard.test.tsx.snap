// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders ProgressIndicatorCard 1`] = `
<div
  class="ds:shadow-border ds:rounded ds:bg-bg-gray-2-25 ds:min-h-[285px] ds:flex ds:items-center ds:justify-center"
  data-testid="pic"
>
  <svg
    aria-busy="true"
    class="ds:animate-spin ds:absolute"
    fill="none"
    height="64"
    role="alert"
    viewBox="0 0 24 24"
    width="64"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      class="ds:fill-accent"
      clip-rule="evenodd"
      d="M0 12.0466C0 12.0466 0 12.0466 0 12.0466C0 5.39345 5.37258 0 12 0C18.6274 0 24 5.39345 24 12.0466C24 18.1898 19.4195 23.259 13.5 24V20.9566C17.7566 20.2397 21 16.5234 21 12.0466C21 7.05674 16.9706 3.01165 12 3.01165C7.02944 3.01165 3 7.05674 3 12.0466C3 12.0466 3 12.0466 3 12.0466H0Z"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;
